JetBrains: add a separate setting for the (optional) dotcom access token

Gedochao opened the pull request:
![image](https://github.com/sourcegraph/sourcegraph/assets/18601388/7b5014a7-aa37-4320-942b-43f545feee8a)

https://github.com/sourcegraph/sourcegraph/assets/18601388/ae45b022-887d-4f3f-83a5-6802e7cd0186
The video doesn't cover all the test scenarios, as I'd have to edit the token string out to cover everything.

## Test plan
(I've tested these things locally, but it is generally advised to double check on another machine, since we don't have UI tests set-up)
- the dotcom/enterprise radio should enable/disable the contained settings
- plugin settings should save both dotcom & enterprise access token between restarts (regardless which one is currently set to be used)
- the `SRC_ACCESS_TOKEN` env var should still override either token for now
- Cody should use the currently set token when calling the API (and not the other)
- the "no access token set" notification should show if the access token isn't set for the current instance type
- validation should work the same for both access token text fields
- there might be some other edge cases, feel free to try and break it

vdavid commented on client/jetbrains/src/main/java/com/sourcegraph/config/SourcegraphProjectService.java:
```diff
@@ -14,7 +14,8 @@
     implements PersistentStateComponent<SourcegraphProjectService>, SourcegraphService {
   @Nullable public String instanceType;
   @Nullable public String url;
-  @Nullable public String accessToken;
```
Wait. I think this is not backwards compatible.

We have existing users who will have `accessToken` set in one of the three levels where this can be set. Unfortunately, we need to keep this as a fallback for `enterpriseAccessToken`. Please mark it as deprecated with some reasoning about this. Then we'll probably need to keep this around for two years or so. 🙈

Gedochao replied:
done, both here and in `SourcegraphApplicationService`.

vdavid reviewed (changes requested):
Requesting three changes:
- Ensure backward compatibility (see my comment)
- Update the readme
- Add changelog item

Otherwise, great stuff!

Thanks!

Gedochao commented:
@vdavid 
- rebased on top of the most recent `main`
- updated `README.md` & `CHANGELOG.md`
- added back the `accessToken` config field for both `SourcegraphApplicationService` and `SourcegraphProjectService`
  - to test locally:
    - the access token for the enterprise instance configured before these changes took place (with the code from the `main` branch) should be respected now, even if it hasn't been reconfigured
    - reconfiguring the access token with these changes should use the new fields
    - reverting back to an older version should also revert back to using the values from the deprecated `accessToken` field
